integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/RoutingBridgeBaseIT.java

```diff
@@ -0,0 +1,736 @@
+					} );
+				} )
+				.setup( IndexedEntity.class ) )
+				.isInstanceOf( SearchException.class )
+				.hasMessageMatching( FailureReportUtils.buildFailureReportPattern()
+						.typeContext( IndexedEntity.class.getName() )
+						.failure( "The binder did not declare any dependency to the entity model during binding."
+								+ " Declare dependencies using context.dependencies().use(...) or,"
+								+ " if the bridge really does not depend on the entity model, context.dependencies().useRootOnly()" )
+						.build() );
+	}
+
+	@Test
+	public void inconsistentDependencyDeclaration() {
+		@Indexed(index = INDEX_NAME)
+		class IndexedEntity {
+			Integer id;
+			String stringProperty;
+			@DocumentId
+			public Integer getId() {
+				return id;
+			}
+			public String getStringProperty() {
+				return stringProperty;
+			}
+		}
+
+		assertThatThrownBy( () -> setupHelper.start()
+				.withConfiguration( b -> {
+					TypeMappingStep typeMapping = b.programmaticMapping().type( IndexedEntity.class );
+					typeMapping.indexed().routingBinder( context -> {
+						// Declare no dependency, but also a dependency: this is inconsistent.
+						context.dependencies()
+								.use( "stringProperty" )
+								.useRootOnly();
+						context.bridge( IndexedEntity.class, new UnusedRoutingBridge<>() );
+					} );
+				} )
+				.setup( IndexedEntity.class ) )
+				.isInstanceOf( SearchException.class )
+				.hasMessageMatching( FailureReportUtils.buildFailureReportPattern()
+						.typeContext( IndexedEntity.class.getName() )
+						.failure( "The binder called context.dependencies().useRootOnly() during binding,"
+								+ " but also declared extra dependencies to the entity model." )
+						.build() );
+	}
+
+	@Test
+	public void useRootOnly() {
+		@Indexed(index = INDEX_NAME)
+		class IndexedEntity {
+			Integer id;
+			String stringProperty;
+			@DocumentId
+			public Integer getId() {
+				return id;
+			}
+			public String getStringProperty() {
+				return stringProperty;
+			}
+		}
+
+		backendMock.expectSchema( INDEX_NAME, b -> { } );
+
+		SearchMapping mapping = setupHelper.start()
+				.withConfiguration( b -> {
+					TypeMappingStep typeMapping = b.programmaticMapping().type( IndexedEntity.class );
+					typeMapping.indexed().routingBinder( context -> {
+						context.dependencies().useRootOnly();
+						context.bridge( IndexedEntity.class,
+								new RoutingBridge<IndexedEntity>() {
+									@Override
+									public void route(DocumentRoutes routes, Object entityIdentifier,
+											IndexedEntity indexedEntity, RoutingBridgeRouteContext context) {
+										routes.addRoute().routingKey( "route/" + entityIdentifier );
+									}
+
+									@Override
+									public void previousRoutes(DocumentRoutes routes, Object entityIdentifier,
+											IndexedEntity indexedEntity, RoutingBridgeRouteContext context) {
+										// The route never changes
+										route( routes, entityIdentifier, indexedEntity, context );
```

Comment:
I thought about it, but decided against it. I expect most users will rely on mutable properties and will need to implement it, so I'd rather make sure they ask themselves "can the previous routes be different?" when implementing the bridge. Especially because failing to implement this correctly will result in duplicate documents in the index.